repair/task_manager_module.hh

```diff
@@ -160,11 +163,46 @@ public:
         return _hints_batchlog_flushed;
     }
 
-    future<> repair_range(const dht::token_range& range, table_id);
+    size_t ranges_size() const noexcept;
+protected:
+    void record_finished_range_repair();
+    future<> run() override;
+
+    virtual future<std::optional<double>> expected_total_workload() const override;
+    virtual std::optional<double> expected_children_number() const override;
+
+    friend class table_repair_task_impl;
+};
+
+class table_repair_task_impl : public repair_task_impl {
+private:
+    shard_repair_task_impl& _local_repair_data;
+    table_id _table_id;
+    uint64_t _ranges_completed = 0;
+public:
+    table_repair_task_impl(tasks::task_manager::module_ptr module,
+            shard_repair_task_impl& shard_task,
+            table_info table,
+            streaming::stream_reason reason)
+        : repair_task_impl(module, tasks::task_id::create_random_id(), 0, shard_task._status.keyspace, table.name, "", shard_task._status.id, reason)
```

Comment:
What are the zero and the one empty string after the table name? I know it is probably sequence_number and entity. Why are them zero and empty? Don't we have at the entity string from the high level task?

Reply:
Yes, they are sequence_number and entity. Sequence number is inherited from parent task so we do not need to set it here. Entity field is empty for all repair tasks

Reply:
Then why do we set sequence_number to  zero?

Also, if we Entity field is empty for all repair tasks, why do we bother to have a parameter for it in repair_task_impl.

Reply:
Top level tasks, i.e. data_sync_repair_task_impl and user_requested_repair_task_impl, which do not have a parent, need to have sequence number set to something meaningful. Thus, we need this parameter and for lower level task it could be anything and 0 is a nice default.

Reply:
We can have a repair_task_impl constructor without the seq and entity parameter.

What are the entity for  data_sync_repair_task_impl and user_requested_repair_task_impl?

Reply:
Yes, entity should definitely be dropped